ansible_collections/arista/avd/roles/eos_designs/templates/network_services/loopback-interfaces.j2

```diff
@@ -2,13 +2,17 @@
 loopback_interfaces:
 {% for tenant in switch.tenants | arista.avd.natural_sort %}
 {%     for vrf in switch.tenants[tenant].vrfs %}
-{%         if tenants[tenant].vrfs[vrf].vtep_diagnostic is defined %}
-  Loopback{{ tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback }}:
+{%         if tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback is arista.avd.defined %}
 {%             set loopback_description = tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback_description | arista.avd.default(vrf ~ '_VTEP_DIAGNOSTICS') %}
+{%             set loopback_ipv4_pool = tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback_ip_range | arista.avd.default(
+                   tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback_ip_pools | arista.avd.default([]) | selectattr('pod', 'arista.avd.defined', pod_name) | map(attribute='ipv4_pool') | first | arista.avd.default) %}
+{%             if loopback_ipv4_pool is arista.avd.defined %}
+  Loopback{{ tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback }}:
     description: {{ loopback_description }}
     shutdown: false
     vrf: {{ vrf }}
-    ip_address: {{ tenants[tenant].vrfs[vrf].vtep_diagnostic.loopback_ip_range | ansible.netcommon.ipaddr('network') | ansible.netcommon.ipmath(switch.id + switch.loopback_ipv4_offset) }}/32
+    ip_address: {{ loopback_ipv4_pool | ansible.netcommon.ipaddr('network') | ansible.netcommon.ipmath(switch.id + switch.loopback_ipv4_offset) }}/32
```

Comment:
We can avoid using ipmath and use only ipaddr, because it better understands the IP addresses, like overflow IP addresses (192.168.0.256) taken care by ipaddr and it returns FALSE.

```
{{ loopback_ipv4_pool | ansible.netcommon.ipaddr(switch.id + switch.loopback_ipv4_offset) | ansible.netcommon.ipaddr('address') }}/32
```

ipaddr('address') will give address only with out mask

Reply:
I think `subnet` might be a better function, but anyway in this PR I do not wish to take on changing the ip addressing schemes. We can do a separate one, when we have tested it thoroughly.

Reply:
You mean this ipsubnet: https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html#subnet-manipulation